### PR TITLE
Sprint 2103

### PR DESCRIPTION
**Pull request by orteliusorthon**
# What has changed
- Styles can now be configured for when a node or edge is both hovered as well as focused
- The onion border can now be configured to appear on both hover and selection (or in tandem)
- The selection mechanism for nodes and edges has been rewritten and moved into a new class (SelectionProcessor). The selection state is now stored on the nodes/edges as well
- Multi selection has been implemented. 
  - Multi selection means that when selecting a node/edge the previous selection is not deselected. 
  - You can toggle the mode on and off by executing a function on the graph. Generally you may want to bind the toggling of this mode to a key (e.g. ctrl). In the example environment I have bound it to ctrl.
- Selection lasso has been implemented. 
  - The selection lasso makes it so that while dragging the mouse over the canvas one can select and entire area where the selection of everything inside should be toggled. 
  - In order to use the lasso you toggle it on and off using a function, similar to the multi select mode. In the case of the example page I have set it to ctrl (same as multi select).
- The onion border color can now be configured
- Cursors can now be configured (still a bit buggy in Chrome unfortunately, but not prioritized)
- The graph will now more accurate set the initial zoom for the graph
- The background grid can now be configured to follow the zoom level.

# Jira Issues
The following issues have been resolved:
- TEAMBADEV-1334 - Implement new style mode for hover + focused
- TEAMBADEV-1335 - Implement setting for onion border on hover
- TEAMBADEV-1301 - Implement node multi selection
- TEAMBADEV-1336 - Implement setting for onion border colour
- TEAMBADEV-1337 - Implement setting for cursors for nodes, edge labels and canvas -- (slightly buggy in Chrome, but not prioritized)
- TEAMBADEV-1339 - Look into initial zoom level for graph
- TEAMBADEV-1338 - Look into setting for how canvas is adjusted upon zoom

### CODE REVIEW

**Review by NickeHa — CHANGES_REQUESTED** [review]
Slår man på enableOnionOnHover :true fungerar selekteringen inte så bra längre.

File: src/js/Config/Env.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ export default Object.freeze({
 
 	//Onion Border
 	ENABLE_ONION_ON_FOCUS: true,
-	ENABLE_ONION_ON_HOVER: false,
```
Comment (NickeHa) [inline src/js/Config/Env.js]: Var det meningen att den ska vara default satt till true?

**Review by NickeHa — CHANGES_REQUESTED** [review]
Jag kan fortfarande inte få selektionen att fungera. Vill du att jag ska visa?

**Review by NickeHa — APPROVED** [review]

**Review by orteliusortdsg — APPROVED** [review]
Review and test approved

**Pull request merged**